grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy

```diff
@@ -149,7 +149,7 @@ class Invoice implements Serializable {
         BigDecimal currentExchangeRate
         String defaultCurrencyCode = Holders.config.openboxes.locale.defaultCurrencyCode
         if (currencyUom?.code != defaultCurrencyCode) {
-            currentExchangeRate = UnitOfMeasureConversion.conversionRateLookup(defaultCurrencyCode, currencyUom?.code).list()
+            currentExchangeRate = UnitOfMeasureConversion.conversionRateLookup(defaultCurrencyCode, currencyUom?.code).get()
```

Comment:
As I saw we can have more than one active conversation rate at the same time (it,s a bit odd 🤔), so I added this sorting by updated date to avoid some different behavior.

Reply:
I agree with @awalkowiak. I think the reason why you can add multiple conversions for the same to/from uoms is because ... lazy. 

For now you can either add uniqueResult = true to the named query. Use the minTransactionDate named query on the TransactionFact domain.

https://github.com/openboxes/openboxes/blob/e19d7df63e4a4b394e7b39000fdb84bf42acfb7b/grails-app/domain/org/pih/warehouse/reporting/TransactionFact.groovy#L53

Or you could do something like this 
```
UnitOfMeasureConversion.conversionRateLookup(defaultCurrencyCode, currencyUom?.code).list()[0]
```
Or this (don't think it's necessary, but who knows.
```
UnitOfMeasureConversion.conversionRateLookup(defaultCurrencyCode, currencyUom?.code).list([max: 1])[0]
```
You'll need to test the result set count = (0, 1, N) cases to make sure this returns an empty list when there are no records and the correct one when there are more than one records.

Reply:
@jmiranda this `get` works as expected.
When there are more than 2 conversion rates, the one with a newer updated date is being taken. When there is no conversion rate the null is returned and it also works as expected.
conversionRateLookup includes order by updated date descending and unique result.

As an example:
I have three conversions from USD to British Pounds
![image](https://github.com/openboxes/openboxes/assets/83239466/97e3a894-5d16-4a78-be8a-92087972976b)
The latest update is this one with a 1.39 conversion rate.
So, going to invoice made in British Pounds
![image](https://github.com/openboxes/openboxes/assets/83239466/b4ba9066-9260-475f-8be8-0533013eb667)
280.78 / 202 = 1.39
So that's the correct conversion rate.

Reply:
Awesome. Sorry I must have missed the uniqueResult = true earlier.